exercises/dominoes/src/test/java/DominoesTest.java

```diff
@@ -0,0 +1,228 @@
+		Dominoes dominoes = new Dominoes();
+		
+		Domino[] dominoesList = {new Domino(1, 2), new Domino(4, 1), new Domino(2, 3)};
+		List<Domino> dominoesArray = new ArrayList<>(Arrays.asList(dominoesList));
+		
+		expectedException.expect(ChainNotFoundException.class);
+		expectedException.expectMessage("No domino chain found.");
+		
+		dominoes.formChain(dominoesArray);
+	}
+	
+	@Ignore("Remove to run test")
+	@Test
+	public void disconnectedSimpleTest() {
+		Dominoes dominoes = new Dominoes();
+		
+		Domino[] dominoesList = {new Domino(1, 1), new Domino(2, 2)};
+		List<Domino> dominoesArray = new ArrayList<>(Arrays.asList(dominoesList));
+		
+		expectedException.expect(ChainNotFoundException.class);
+		expectedException.expectMessage("No domino chain found.");
+		
+		dominoes.formChain(dominoesArray);
+	}
+	
+	@Ignore("Remove to run test")
+	@Test
+	public void disconnectedDoubleLoopTest() {
+		Dominoes dominoes = new Dominoes();
+		
+		Domino[] dominoesList = {new Domino(1, 2), new Domino(2, 1), new Domino(3, 4), new Domino(4, 3)};
+		List<Domino> dominoesArray = new ArrayList<>(Arrays.asList(dominoesList));
+		
+		expectedException.expect(ChainNotFoundException.class);
+		expectedException.expectMessage("No domino chain found.");
+		
+		dominoes.formChain(dominoesArray);
+	}
+	
+	@Ignore("Remove to run test")
+	@Test
+	public void disconnectedSingleIsolatedTest() {
+		Dominoes dominoes = new Dominoes();
+		
+		Domino[] dominoesList = {new Domino(1, 2), new Domino(2, 3), new Domino(3, 1), new Domino(4, 4)};
+		List<Domino> dominoesArray = new ArrayList<>(Arrays.asList(dominoesList));
+		
+		expectedException.expect(ChainNotFoundException.class);
+		expectedException.expectMessage("No domino chain found.");
+		
+		dominoes.formChain(dominoesArray);
+	}
+	
+	@Ignore("Remove to run test")
+	@Test
+	public void needBacktrackTest() {
+		Dominoes dominoes = new Dominoes();
+		
+		Domino[] dominoesList = {new Domino(1, 2), new Domino(2, 3), new Domino(3, 1), new Domino(2, 4), new Domino(4, 2)};
+		ArrayList<Domino> dominoesArray = new ArrayList<>(Arrays.asList(dominoesList));
+		
+		List<Domino> chain = dominoes.formChain(dominoesArray);
+		
+		assertValidChain(dominoesArray, chain);
+	}
+	
+	@Ignore("Remove to run test")
+	@Test
+	public void separateLoopsTest() {
+		Dominoes dominoes = new Dominoes();
+		
+		Domino[] dominoesList = {new Domino(1, 2), new Domino(2, 3), new Domino(3, 1),
+				new Domino(1, 1), new Domino(2, 2), new Domino(3, 3)};
+		ArrayList<Domino> dominoesArray = new ArrayList<>(Arrays.asList(dominoesList));
+		
+		List<Domino> chain = dominoes.formChain(dominoesArray);
+		
+		assertValidChain(dominoesArray, chain);
+	}
+	
+	@Ignore("Remove to run test")
+	@Test
+	public void nineElementsTest() {
+		Dominoes dominoes = new Dominoes();
+		Domino[] dominoesList = {new Domino(1, 2), new Domino(5, 3), new Domino(3, 1), new Domino(1, 2), new Domino(2, 4), new Domino(1, 6), new Domino(2, 3), new Domino(3, 4), new Domino(5, 6)};
```

Comment:
It would be more readable if you add spread this over several lines like you did for the previous test 🙂